src/rebar_app_utils.erl

```diff
@@ -232,9 +232,9 @@ parse_dep(Parent, {Name, Source, Opts}, DepsDir, IsLock, State) when is_tuple(So
     ?WARN("Dependency option list ~p in ~p is not supported and will be ignored", [Opts, Name]),
     dep_to_app(Parent, DepsDir, Name, [], Source, IsLock, State);
 parse_dep(Parent, {Name, {pkg, PkgName, Vsn}, Level}, DepsDir, IsLock, State) when is_integer(Level) ->
-    dep_to_app(Parent, DepsDir, Name, Vsn, {pkg, PkgName, Vsn, undefined}, IsLock, State);
-parse_dep(Parent, {Name, {pkg, PkgName, Vsn, Hash}, Level}, DepsDir, IsLock, State) when is_integer(Level) ->
-    dep_to_app(Parent, DepsDir, Name, Vsn, {pkg, PkgName, Vsn, Hash}, IsLock, State);
+    dep_to_app(Parent, DepsDir, Name, Vsn, {pkg, PkgName, Vsn, undefined, undefined}, IsLock, State);
+parse_dep(Parent, {Name, {pkg, PkgName, Vsn, OldHash,  Hash}, Level}, DepsDir, IsLock, State) when is_integer(Level) ->
+    dep_to_app(Parent, DepsDir, Name, Vsn, {pkg, PkgName, Vsn, OldHash, Hash}, IsLock, State);
```

Comment:
don't we have a potential case for `OldHash =/= undefined` with also `Hash =:= undefined` ?

Reply:
I thought we did, but now I'm not so sure. Running rebar3 will acquire and write the new locks, along with the old. Given that and if you've upgraded to this version of rebar3, there should be no need to take check against the old hash. Make sense?

Reply:
I was thinking of what happens if you load an old version of a lock on your first time with the new registry?

Reply:
It works. A new lockfile will be generated with old locks and new locks.

Reply:
Sounds good then. I didn't think they'd be there at the same time but I guess by that point they have been resolved already.

Reply:
As discussed on slack, we will go ahead and add in some logic to check for the inner checksum if the check against the outer checksum fails for good measure.